src/Features/Core/Portable/AddImport/AbstractAddImportFeatureService.cs

```diff
@@ -228,11 +228,12 @@ private static bool IsHostOrRemoteWorkspace(Project project)
                 source: viableUnreferencedProjects,
                 produceItems: static async (project, onItemsFound, args, cancellationToken) =>
                 {
+                    var (projectToAssembly, allSymbolReferences, maxResults, finder, exact, linkedTokenSource) = args;
```

Comment:
the pattern is to take the args parameter (which here was written as `args: (projectToAssembly, allSymbolReferences, maxResults, finder, exact, linkedTokenSource),` and just take that full tuple and copy it into the variable as what we are dconstructing into.  Then, within the body, we can refer to all the nice names again instead of hte `args.XXX` in every single place.

Reply:
if we follow this pattern in the future, it will also make diffs easier, as the body code which got pushed into the lambda will stay the same.